### PR TITLE
feat(sourcemaps): Implement support for date last modified in the upload and endpoints

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR adds a new last_modified_date field to artifact bundles

### CODE REVIEW

File: src/sentry/models/artifactbundle.py [hunk 1]
```diff
@@ -53,9 +53,14 @@ class ArtifactBundle(Model):
...
     artifact_count = BoundedPositiveIntegerField()
+    # This field represents the date in which the bundle was renewed, since we have a renewal mechanism in place. The
+    # name is the same across entities connected to this bundle named *ArtifactBundle.
     date_added = models.DateTimeField(default=timezone.now, db_index=True)
-    # This field represents the date of the upload that we show in the UI.
+    # This field represents the date of upload of this bundle, and it's not mutated afterward.
     date_uploaded = models.DateTimeField(default=timezone.now)
+    # This field represents the date in which this bundle was last modified, where modification means that an
+    # association has been added or any of its fields have been modified.
+    date_last_modified = models.DateTimeField(default=timezone.now, null=True)
```
Comment (Swatinem) [inline src/sentry/models/artifactbundle.py:63]: The comments help, but the naming in itself is still extremely confusing.

**Review by Swatinem — APPROVED** [review]

File: src/sentry/models/artifactbundle.py (hunk 1 above)
Comment (iambriccardo) [reply src/sentry/models/artifactbundle.py:63]: Naming of which field? I agree it's hella confusing but changing column names is a pain in the ass.

Comment (Swatinem) [reply src/sentry/models/artifactbundle.py:63]: Well, `date_added` is confusing, as it reflects the time when it was last touched in terms of auto-expiration.
`date_last_modified` would be a better name for *that*. However, it also conflicts with the concept of associations that you are now adding.

How do those two things relate? As in: What happens to an association if a bundle is auto-expired?

Comment (iambriccardo) [reply src/sentry/models/artifactbundle.py:63]: Cascading deletions are programmed to happen.

**Review by Swatinem — APPROVED** [review]

**Pull request merged**